scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,525 @@
+    var dbOptions = {};
+    dbOptions.couchDbUrl = processArgv[2];
+    dbOptions.staticDataDir = processArgv[3];
+    dbOptions.buildDataDir = processArgv[4];
+    if (processArgv.length > 5 && processArgv[5] === "--justDelete") { // for debugging.
+        dbOptions.justDelete = true;
+    } else {
+        dbOptions.justDelete = false;
+    }
+
+    // Set up database specific options
+    dbOptions.viewsUrl = dbOptions.couchDbUrl + "/_design/views";
+    dbOptions.prefsSafesViewUrl = dbOptions.couchDbUrl + "/_design/views/_view/findSnapsetPrefsSafes";
+    dbOptions.gpiiKeysViewUrl = dbOptions.couchDbUrl + "/_design/views/_view/findAllGpiiKeys";
+    dbOptions.parsedCouchDbUrl = url.parse(dbOptions.couchDbUrl);
+    dbOptions.staticData = [];
+    /* dbOptions.newViews; */
+    /* dbOptions.oldViews; */
+    dbOptions.snapsetPrefsSafes = [];
+    dbOptions.gpiiKeys = [];
+    dbOptions.postOptions = {
+        hostname: dbOptions.parsedCouchDbUrl.hostname,
+        port: dbOptions.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        auth: dbOptions.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // IMPORTANT: FILL IN PER REQUEST
+            "Content-Type": "application/json"
+        }
+    };
+    fluid.log("COUCHDB_URL: '" +
+        dbOptions.parsedCouchDbUrl.protocol + "//" +
+        dbOptions.parsedCouchDbUrl.hostname + ":" +
+        dbOptions.parsedCouchDbUrl.port +
+        dbOptions.parsedCouchDbUrl.pathname + "'"
+    );
+    fluid.log("STATIC_DATA_DIR: '" + dbOptions.staticDataDir + "'");
+    fluid.log("BUILD_DATA_DIR: '" + dbOptions.buildDataDir + "'");
+
+    return dbOptions;
+};
+
+/*
+ * Loads the static data from disk, and creates a separate reference to the
+ * views document.
+ * @param {Object} options - Object that has the path to the directory
+ *                           containing the static data. On output, contains
+ *                           members for the new views and other static data.
+ */
+gpii.dataLoader.loadStaticDataFromDisk = function (options) {
+    var data = gpii.dataLoader.getDataFromDirectory(options.staticDataDir);
+    var views = fluid.find(data, function (anElement) {
+        if (anElement._id && anElement._id === "_design/views") {
+            return anElement;
+        } else {
+            return undefined;
+        }
+    });
+    options.staticData = data;
+    options.newViews = views;
+    fluid.log("Retrieved static data from: '" + options.staticDataDir + "'");
+    fluid.log("\tViews data " + ( views ? "retrieved." : "missing." ));
+};
+
+/*
+ * Utility to configure a step:  creates a response callback, binds it to an
+ * http database request, and configures a promise to resolve/reject when the
+ * response callback finishes or fails.
+ * @param {Object} details - Specific information for the request and response,
```

Comment:
Document mandatory fields in this object either inline here or else using a JSDocs @typedef http://usejsdoc.org/tags-typedef.html if this options structure appears in the signature of other functions

Reply:
I've defined a few @typedef and @callback block tags and used them accordingly throughout.   And, where `options` is a parameter, I've followed the [Parameters with properties](http://usejsdoc.org/tags-param.html#parameters-with-properties) JSDoc section, listing the properties of `options` specifically used within the function.

That said, this function, `gpii.dataLoader.configureStep()`, was a struggle because it is abstract and generic.  The actual option properties and their meaning is defined in the specific "set up" functions that ultimately call this one.  It's tricky being informative and vague at the same time...